packages/@react-spectrum/breadcrumbs/src/Breadcrumbs.tsx

```diff
@@ -132,38 +132,39 @@ function Breadcrumbs<T>(props: SpectrumBreadcrumbsProps<T>, ref: DOMRef) {
 
   useLayoutEffect(updateOverflow, [children, updateOverflow]);
 
-  useLayoutEffect(() => {
-    // Only try to restore focus after an action.
-    if (!isAfterAction.current) {
-      return;
-    }
-    isAfterAction.current = false;
+  // useLayoutEffect(() => {
+  //   // Only try to restore focus after an action.
+  //   if (!isAfterAction.current) {
+  //     return;
+  //   }
+  //   isAfterAction.current = false;
 
-    // Wait a frame for updateOverflow.
-    requestAnimationFrame(() => {
-      if (
-        // Where breadcrumbs exist, 
-        domRef.current &&
-        // the current breadcrumb is defined,
-        currentRef.current &&
-        // and the current focus has either been removed from the DOM,
-        // or is within the breadcrumbs,
-        (
-          document.activeElement === document.body ||
-          domRef.current.contains(document.activeElement)
-        )
-      ) {
-        // focus the current breadcrumb.
-        currentRef.current.UNSAFE_getDOMNode().focus();
-      }
-    });
-  }, [domRef, currentRef, isAfterAction, children]);
+  //   // Wait a frame for updateOverflow.
+  //   requestAnimationFrame(() => {
+  //     if (
+  //       // Where breadcrumbs exist,
+  //       domRef.current &&
+  //       // the current breadcrumb is defined,
+  //       currentRef.current &&
+  //       // and the current focus has either been removed from the DOM,
+  //       // or is within the breadcrumbs,
+  //       (
+  //         document.activeElement === document.body ||
+  //         domRef.current.contains(document.activeElement)
+  //       )
+  //     ) {
+  //       // focus the current breadcrumb.
+  //       currentRef.current.UNSAFE_getDOMNode().focus();
+  //     }
+  //   });
+  // }, [domRef, currentRef, isAfterAction, children]);
 
   function triggerOnAction(key: Key) {
     if (onAction) {
       onAction(key);
-      isAfterAction.current = true;
+      // isAfterAction.current = true;
     }
+    currentRef.current.UNSAFE_getDOMNode().focus();
```

Comment:
I think it makes sense to always focus the current breadcrumb immediately after selecting an item, rather than waiting for a re-render.

Reply:
This won't work after selecting a breadcrumb from the dropdown menu, we need to wait a frame.

Reply:
It should work. See below comment.